### PR TITLE
feat: Adjustments to Generic Business Partner model

### PR DESCRIPTION
**Pull request by martinfkaeser**
Small improvements to the DTO model:
- optional BPN fields are added for input
- named bpnL/S/A instead of bpnl/s/a
- fixed assigment for isOwner, addressType after changes to optional/mandatory
- improved OpenAPI description (clarified sorting and de-duping behaviour)

Solves #384

### CODE REVIEW

**Review by nicoprow — APPROVED** [review]
looks good to me

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt [hunk 1]
```diff
@@ -60,16 +63,16 @@ class BusinessPartnerMappings {
...
             states = entity.states.map(::toStateDto),
             classifications = entity.classifications.map(::toClassificationDto),
             roles = entity.roles,
             postalAddress = toPostalAddressOutputDto(entity.postalAddress),
-            isOwner = entity.isOwner ?: false,      // TODO should be mandatory in entity
-            bpnl = entity.bpnL ?: throw NullPointerException("bpnL is null"),
-            bpns = entity.bpnS,
-            bpna = entity.bpnA ?: throw NullPointerException("bpnA is null"),
+            isOwner = entity.isOwner,
+            bpnL = entity.bpnL ?: throw NullPointerException("bpnL is null"),
```
Comment (rainer-exxcellent) [inline bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt:73]: Shouldn't we throw a Bpdm...Exception here?

File: bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt [hunk 2]
```diff
@@ -30,27 +30,36 @@ interface IBaseBusinessPartnerDto {
...
-    @get:Schema(description = "")
+    @get:ArraySchema(arraySchema = Schema(description = "The list of name parts to accommodate the different number of name fields in different systems."))
     val nameParts: List<String>
 
-    @get:Schema(description = "Abbreviated name or shorthand")
+    @get:Schema(description = "Abbreviated name or shorthand.")
     val shortName: String?
 
-    @get:ArraySchema(arraySchema = Schema(description = "The list of identifiers of the business partner."))
+    @get:ArraySchema(arraySchema = Schema(description = "The list of identifiers of the business partner. Sorted and duplicates are removed."))
```
Comment (rainer-exxcellent) [inline bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt:39]: This comment is used for request and response. 
The request does not have to be sorted.
Duplicates are only removed in the request. The response should not contain any duplicate entries

Comment (martinfkaeser) [reply bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt:39]: I tried to phrase it that it works for both the request and response. 
But yes, it's not quite clear. What do you think about:
"Automatically sorted and duplicates removed."
Any other suggestions?

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt (hunk 1 above)
Comment (martinfkaeser) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt:73]: You mean like that:
`throw BpdmNullMappingException(BusinessPartner::class, BusinessPartnerOutputDto::class, BusinessPartnerOutputDto::bpnL, entity.externalId)`

File: bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt (hunk 2 above)
Comment (martinfkaeser) [reply bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt:39]: Idea: "Sorted and duplicates removed by the service."

Comment (nicoprow) [reply bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/IBaseBusinessPartnerDto.kt]: I agree, sounds better to me

**Pull request merged**